examples/plot_investigating_dirty_categories.py

```diff
@@ -122,24 +123,51 @@
 f2.tight_layout()
 
 ########################################################################
-# As shown in the previous plot, we see that "communication Equipment technician"'s
-# nearest neighbor is "telecommunication technician", although it is also
+# As shown in the previous plot, we see that the nearest neighbor of
+# "Communication Equipment Technician"
+# is "telecommunication technician", although it is also
 # very close to senior "supply technician": therefore, we grasp the
 # "communication" part (not initially present in the category as a unique word)
 # as well as the technician part of this category.
 
 
-
 #########################################################################
 # Encoding categorical data using SimilarityEncoder
 # -------------------------------------------------
 #
 # A typical data-science workflow uses one-hot encoding to represent
 # categories.
+from sklearn.preprocessing import CategoricalEncoder
+
+n_obs, n_categories = 20, 20
+# employee_position_titles = values[
```

Comment:
Right now, we use unique sorted values as entries to the one hot encoder, to the transformed values 
are the identity matrix.
the 4 commented lines pick observations from non-unique non sorted values at random, so we will get
a more realistic one-hot encoded martrix (the columns won't necessarily sum to 1, and the 'hot' values will be scattered everywhere in the matrix, not only in the diagonal)
Thoughts @GaelVaroquaux?

Reply:
I don't think that we should use the sorted values: it does not reflect the usage. We should fit-transform a "head" of the data matrix.